packages/router/src/grpc/view-protocol-server/nullifier-status.ts

```diff
@@ -29,5 +28,18 @@ export const handleNullifierStatusReq = async (
   const noteSpent = noteByNullifier ? noteByNullifier.heightSpent !== 0n : false;
   const swapSpent = swapByNullifier ? swapByNullifier.heightClaimed !== 0n : false;
 
-  return new NullifierStatusResponse({ spent: noteSpent || swapSpent });
+  if (noteSpent || swapSpent) {
+    return new NullifierStatusResponse({ spent: true });
+  } else if (!req.awaitDetection) {
+    return new NullifierStatusResponse({ spent: false });
+  }
```

Comment:
There's an inconsistency here with the rust view service, I missed it when I implemented this method. 
The thing is that rust just ignores nullifiers from swap, so we should ignore them too.

Reply:
That might be a bug in the rust view server?

Reply:
I'm not sure, but I think it would generally be useful to track swap nullifier to make sure the swap is claimed.

Reply:
But at the same time, the response of this method is `spent: true/false` but swap nullifier cannot be spent. 
So I'm leaning towards the fact that there is no bug in rust, and this method is only responsible for getting the status of note nullifier only

Reply:
Indeed, the Rust code is not looking in the swaps table it seems:

https://github.com/penumbra-zone/penumbra/blob/main/crates/view/src/storage.rs#L510-L519

Reply:
Also, NullifierStatusResponse says "spent", but I suppose the consumer would know that in the case of swaps, that would mean "claimed"?

Reply:
The SwapClaim "spends" the nullifier in the sense that it consumes it and it can't be claimed again. Since we moved to generic nullifiers (nullifiers are generated for arbitrary state commitments, notes, swaps, etc) "spending" a nullifier refers to consuming it, generally. I'd say this is a bug in the Rust view server and it probably causes problems.